Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean

```diff
@@ -714,6 +714,98 @@ theorem mfderiv_prod_eq_add_apply {f : M × M' → M''} {p : M × M'} {v : Tange
 
 end Prod
 
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {n : WithTop ℕ∞}
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E'] {H' : Type*} [TopologicalSpace H']
+  {J : Type*} {J : ModelWithCorners 𝕜 E' H'}
+  {N N' : Type*} [TopologicalSpace N] [TopologicalSpace N'] [ChartedSpace H' N] [ChartedSpace H' N']
+
+open Topology
+
+variable {f : M → N} (g : M' → N') {s : Set M} {t : Set M'} {p : M} {q : M'}
+
+theorem hasMFDerivWithinAt_inl :
+    HasMFDerivWithinAt I I (@Sum.inl M M') s p
+      (ContinuousLinearMap.id 𝕜 (TangentSpace I p)) := by
+  constructor
+  · exact continuous_inl.continuousWithinAt
+  · set U := (extChartAt I p).symm ⁻¹' s ∩ Set.range I
+    set x0 := (extChartAt I p) p
+    have h_eventually_range :
```

Comment:
You can write the same result in a slightly simpler way:
- if you leave out the y argument, writtenInExtChart is already a function, so just writing `writtenInExtChartAt I I p (@Sum.inl M M')` on the left hand side works already
- on the right hand side, you have just `id`. (Lean can guess the type from the left hand side; you need not specify it.)
All in all, I would write this as:
```
    have h_eventually_range :
        writtenInExtChartAt I I p (@Sum.inl M M') =ᶠ[𝓝[Set.range I] x₀] id := by

Reply:
Also, this statement looks like it might be re-usable: cna you extract this as a separate lemma? Thank you!